familias/serializers.py

```diff
@@ -0,0 +1,280 @@
+        return Tutor.objects.create(**self.validated_data)
+
+    def update(self):
+        """ This function overides the default behaviour for creating
+            an object through a serializer since Nested Updates are
+            not implemented by DRF.
+
+            Returns
+            -------
+            Updated Instance of Tutor model.
+        """
+        Tutor.objects.filter(pk=self.instance.pk).update(**self.validated_data)
+        return Tutor.objects.get(pk=self.instance.pk)
+
+
+class IntegranteSerializer(serializers.ModelSerializer):
+    """ Serializer to represent a .models.Integrante instance
+        through a REST endpoint for the offline application
+        to submit information.
+
+        DRF by default sets the id field to read_only. This
+        makes it unaccesible in validated_data. When an instance
+        is being updated we need to acces the id in the nested
+        serializers so that we can send the actual instance to
+        the update method.
+    """
+    alumno_integrante = AlumnoSerializer(allow_null=True)
+    tutor_integrante = TutorSerializer(allow_null=True)
+
+    class Meta:
+        model = Integrante
+        fields = (
+            'id',
+            'nombres',
+            'apellidos',
+            'telefono',
+            'correo',
+            'nivel_estudios',
+            'fecha_de_nacimiento',
+            'alumno_integrante',
+            'tutor_integrante',
+            'activo')
+
+        extra_kwargs = {"id": {"read_only": False, "required": False}}
+
+    def create(self, family):
+        """ This function overides the default behaviour for creating
+            an object through a serializer.
+
+            This serializer's object dependes on a .models.Integrante
+            object instance. Since we are dealing with nested
+            created objects, the integrante must be created first
+            and passed as parameter to the created function.
+
+            The Integrante model has a two sepcialization which are
+            indicated in the dictionary data as 'alumno_integrante'
+            and 'tutor_integrante'. This means the Integrante is
+            either a Tutor or an Alumno.
+
+            This function removes that data from the validated_data
+            dictionary, creates the Integrante and then creates the
+            specialized instance of either Alumno or Tutor.
+
+            Notes
+            -----
+            save_foreign_relationship does not create object on None
+            object. Therefore if one of the specilizations is not
+            indicated it is not created.
+        """
+        self.validated_data['familia'] = family
+
+        alumno = self.validated_data.pop('alumno_integrante')
+        tutor = self.validated_data.pop('tutor_integrante')
+
+        integrante = Integrante.objects.create(**self.validated_data)
+
+        save_foreign_relationship([alumno], AlumnoSerializer, Alumno, integrante)
```

Comment:
We talked about it on Thursday with @AndreeToledo, that we should validate this in the model. Although there should also a exist a validation here so that in the catastrophic case it were to happen it was properly handled.